### ISSUE TITLE
Always compute version number and pass it in context.

### ISSUE THREAD

**Issue by rasa** [kdeldycke/click-extra#325]
Given
```python
import click_extra
@click_extra.extra_command(params=[
  click_extra.VersionOption(version="0.1")
])
def cli():
  print("%s version %s" % (__file__, ?))

cli()
```
what do I need to replace `?` with? Note that I scanned the code for close to an hour, before giving up.

**Comment by kdeldycke** [comment]
As a reminder, the current version option in Click Extra is a full copy of Click's original code. But made into a class here, to allow its use with the declarative `params=` argument. Which [fixes Click #2324 issue](https://github.com/pallets/click/issues/2324).

So in its current capacity the version string is not exposed. It is detected or consumed right before exit the CLI execution flow, as you can see here:

https://github.com/kdeldycke/click-extra/blob/fa9b002a9d7fda150916efcdab967a6b2264a8dc/click_extra/version.py#L124-L125

To have it support what you're looking for, we need to refactor the version code and enhance it. This is a feature request.

**Comment by kdeldycke** [comment]
BTW, I used your issue as an excuse to bootstrap the version option documentation at: https://kdeldycke.github.io/click-extra/version.html

**Comment by kdeldycke** [comment]
I just realized that this is something that was more or less requested by @puzzlepeaches in #319. The difference with the later is that the `--config` parameter is provided on the command line, so there is an [alternative way to gather this raw, unprocessed value](https://kdeldycke.github.io/click-extra/commands.html#introspecting-parameters).

I guess this workaround was good enough in that case, but doesn't cut it for the `--version` parameter.

All in all, there is an opportunity somewhere to have all processed parameters be available in the context.

**Comment by kdeldycke** [comment]
Another thing to consider is the way I keep the table format ID in the context. See: https://github.com/kdeldycke/click-extra/commit/9121c6fd692eca7133cb00307cfc535198a3b0e9

**Comment by kdeldycke** [comment]
Good news @rasa, you can now fetch version's metadata from the context. Everything's explained in the documentation: file:///Users/kde/click-extra/docs/html/version.html#get-metadata-values

It is part of the `main` branch and will be available in the upcoming Click Extra v4.5.0 release.

**Comment by kdeldycke** [comment]
Click Extra 4.5.0 has just been released, and now expose all sorts of version-related metadata in the context: https://kdeldycke.github.io/click-extra/version.html#get-metadata-values
